src/source/source_cache.js

```diff
@@ -870,8 +926,10 @@ class SourceCache extends Evented {
 SourceCache.maxOverzooming = 10;
 SourceCache.maxUnderzooming = 3;
 
-function compareKeyZoom(a, b) {
-    return ((a % 32) - (b % 32)) || (b - a);
+function compareTileId(a: OverscaledTileID, b: OverscaledTileID): number {
+    const aWrap = Math.abs(a.wrap * 2) - +(a.wrap < 0);
+    const bWrap = Math.abs(b.wrap * 2) - +(b.wrap < 0);
```

Comment:
Can you explain the `wrap` sorting logic? I can't quite get what's going on here...

Reply:
Different copies of the world are sorted based on their distance to the center where wrap is 0. This implementation converts wrap values to unsigned distances by reserving odd numbers for copies with negative wrap and even numbers for copies with positive wrap. The sorted order of wrap values would be: 0, -1, 1, -2, 2, -3, 3 and so on.

This is effectively the same as 
```
    wrap *= 2;
    if (wrap < 0) wrap = wrap * -1 - 1;
```

Reply:
Oh, so like zig-zag encoding. Gotcha. Maybe worth adding a comment since it's not obvious.